dashboard/views_test.py

```diff
@@ -344,3 +347,45 @@ def test_enrollment(self, mock_refresh, mock_edx_enr, mock_index):  # pylint: di
         cache_enr = CachedEnrollment.objects.filter(
             user=self.user, course_run__edx_course_key=self.course_id).first()
         assert cache_enr is not None
+
+
+class UserPaymentStatusTest(MockedESTestCase, APITestCase):
+    """
+    Tests for the UserCourseEnrollment REST API
+    """
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        # create an user
+        cls.user = UserFactory.create()
+        cls.program = ProgramFactory.create()
+        cls.url = reverse('user_program_payment')
+
+    def setUp(self):
+        super().setUp()
+        self.client.force_login(self.user)
+
+    def test_enrolled_and_paid_user(self):
+        """Enrolled and paid in any course will get True"""
+        ProgramEnrollment.objects.create(user=self.user, program=self.program)
+        CachedEnrollmentFactory.create(
+            user=self.user,
+            course_run__course__program=self.program,
+            verified=True
+        )
+
+        resp = self.client.get(self.url, format='json')
+        assert len(resp.data) == 1
+        assert resp.data[0][str(self.program.id)] is True
+
+    def test_not_enrolled_and_paid_user(self):
+        """when user is enrolled in program but not in any course"""
+        ProgramEnrollment.objects.create(user=self.user, program=self.program)
```

Comment:
i am using `ProgramEnrollment` here, because the Factory also creates cache objects which marks user paid